core/src/banking_stage.rs

```diff
@@ -3,88 +3,66 @@
-    solana_metrics::inc_new_counter_info,
-    solana_perf::{
-        data_budget::DataBudget,
-        packet::{Packet, PacketBatch, PACKETS_PER_BATCH},
+        unprocessed_transaction_storage::{ThreadType, UnprocessedTransactionStorage},
     },
-    solana_poh::poh_recorder::{BankStart, PohRecorder, PohRecorderError, TransactionRecorder},
-    solana_program_runtime::timings::ExecuteTimings,
+    crossbeam_channel::{Receiver as CrossbeamReceiver, Sender as CrossbeamSender},
+    solana_client::connection_cache::ConnectionCache,
+    solana_gossip::cluster_info::ClusterInfo,
+    solana_ledger::blockstore_processor::TransactionStatusSender,
+    solana_perf::{data_budget::DataBudget, packet::PacketBatch},
+    solana_poh::poh_recorder::{PohRecorder, PohRecorderError},
     solana_runtime::{
-        accounts::TransactionLoadResult,
-        bank::{
-            Bank, CommitTransactionCounts, LoadAndExecuteTransactionsOutput,
-            TransactionBalancesSet, TransactionCheckResult, TransactionExecutionResult,
-            TransactionResults,
-        },
-        bank_forks::BankForks,
-        bank_utils,
-        transaction_batch::TransactionBatch,
-        transaction_error_metrics::TransactionErrorMetrics,
+        self, bank_forks::BankForks, transaction_error_metrics::TransactionErrorMetrics,
         vote_sender_types::ReplayVoteSender,
     },
     solana_sdk::{
-        clock::{
-            Slot, DEFAULT_TICKS_PER_SLOT, FORWARD_TRANSACTIONS_TO_LEADER_AT_SLOT_OFFSET,
-            HOLD_TRANSACTIONS_SLOT_OFFSET, MAX_PROCESSING_AGE,
-        },
-        feature_set::allow_votes_to_directly_update_vote_state,
-        pubkey::Pubkey,
-        saturating_add_assign,
-        timing::{duration_as_ms, timestamp, AtomicInterval},
-        transaction::{self, SanitizedTransaction, TransactionError, VersionedTransaction},
-        transport::TransportError,
-    },
-    solana_streamer::sendmmsg::batch_send,
-    solana_transaction_status::{
-        token_balances::TransactionTokenBalancesSet, TransactionTokenBalance,
+        feature_set::allow_votes_to_directly_update_vote_state, pubkey::Pubkey,
+        timing::AtomicInterval,
     },
+    solana_transaction_status::TransactionTokenBalance,
     std::{
         cmp,
         collections::HashMap,
         env,
-        net::{SocketAddr, UdpSocket},
+        net::UdpSocket,
         sync::{
             atomic::{AtomicU64, AtomicUsize, Ordering},
             Arc, RwLock,
         },
         thread::{self, Builder, JoinHandle},
-        time::{Duration, Instant},
+        time::Duration,
     },
 };
 
+pub mod commit_executor;
+pub mod consume_executor;
+pub mod decision_maker;
+pub mod external_scheduler;
+pub mod forward_executor;
+pub mod multi_iterator_scheduler;
+pub mod packet_receiver;
+pub mod record_executor;
+pub mod scheduler_error;
+pub mod scheduler_handle;
+pub mod thread_aware_account_locks;
+pub mod thread_local_scheduler;
```

Comment:
name shedding: it seems both `*er` and `*_executor` is used? Is there some criteria of using one over the other? I prefer `*er` to `*_executor` for its shortness. 

so, `commiter` and `consumer`, etc...

Reply:
Probably not as careful about it as I should have been with naming, but attempted to keep `er`s as components which made logical decisions, or passed new data to another component (decision making, packet receiving, scheduling), where `executor`s were intended to process data in some way and return some result to the calling component.

I'm also fine, and think it's probably less jarring/confusing to just rename these all to `er` as well.

Reply:
i sensed that  overall naming is well thought-out and helpful to navigate this 4k loc patch. :) i just nit-picking here. ;)